mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1327,7 +1327,9 @@ static Type typeFromString(StringRef name, MLIRContext *ctx) {
   std::optional<Type> result =
       llvm::StringSwitch<std::optional<Type>>(name)
           .Case("f32", Float32Type::get(ctx))
+          .Case("fp32", Float32Type::get(ctx))
```

Comment:
When I was thinking of merging typeFromString() and strToType() -- the latter includes these two cases.